events/func.py

```diff
@@ -2,34 +2,105 @@
 import os
 import json
 import logging
-import requests
+
+import asyncio
+import fdk
+import oci
 
 from fdk import response
+from slack import WebClient
+from slack.errors import SlackApiError
 
 local_dev = os.environ.get('is_local')
+compartment_id = os.environ.get('compartment_id')
+slack_bot_token = os.environ.get('slack_bot_token')
+
+FORMAT = '%(asctime)s -- %(message)s'
+logging.basicConfig(format=FORMAT)
 logger = logging.getLogger()
 
+TABLE_NAME = "slack_events"
+
+slack_client = WebClient(token=slack_bot_token, run_async=True)
+
+
+def should_post_message(oci_client, event):
+    # Write to DB to block other functions
+    # If write fails event already handled
+    should_post = update_db(oci_client, event)
+
+    return should_post
+
+
+def update_db(oci_client, event):
+    updated = False
+
+    try:
+        row = oci.nosql.models.UpdateRowDetails(
+            compartment_id=compartment_id,
+            option="IF_ABSENT",
+            is_get_return_row=True,
+            value={
+                "event_type": event["type"],
+                "channel_id": event["channel"],
+                "user_id": event["user"],
+                "data": event
+            })
+
+        response = oci_client.update_row(TABLE_NAME, update_row_details=row)
 
-def handle_member_join_channel(event):
+        if response.data.existing_version is None:
+            updated = True
+    except Exception as ex:
+        # TODO: if fail to write to DB alarm
+        logger.error(f"db write error: {str(ex)}")
+
+    return updated
+
+
+async def handle_member_join_channel(event):
     random_channel_id = os.environ.get('random_channel_id')
+    if (event["channel"] != random_channel_id):
+        return
+
+    oci_client = None
+    if local_dev:
+        config = oci.config.from_file("./oci_config")
+        oci_client = oci.nosql.NosqlClient(config)
+    else:
+        signer = oci.auth.signers.get_resource_principals_signer()
+        oci_client = oci.nosql.NosqlClient({}, signer=signer)
```

Comment:
nit: This could be made into a function and the results saved to oci_client. Something along the lines of 
`oci_client = init_client(local_dev)`.  additional nit: the client is a nosql client so `nosql_client` or `db_client` might help someone quickly understand what the client is doing.

Reply:
Fixed